src/state/session/index.tsx

```diff
@@ -102,10 +102,21 @@ function createPersistSessionHandler(
     expired: boolean
     refreshedAccount: SessionAccount
   }) => void,
+  {
+    unstable_networkErrorCallback,
+  }: {
+    unstable_networkErrorCallback?: () => void
+  } = {},
 ): AtpPersistSessionHandler {
   return function persistSession(event, session) {
     const expired = event === 'expired' || event === 'create-failed'
 
+    if (event === 'network-error') {
+      logger.warn(`session: persistSessionHandler received network-error event`)
+      unstable_networkErrorCallback?.()
+      return
+    }
```

Comment:
Only real addition is this block, just kicks the user out to signup and sends a warning to sentry

Reply:
Why in a sub-object with the unstable_ prefix?

Reply:
Oh, just removed, was just what I called it initially